### ISSUE TITLE
Different values for same loss and eval metric (RMSE) on same data with categorical features in model fitting

### ISSUE THREAD

**Issue by Fersol** [catboost/catboost#1799]
catboost version: 0.26
Operating System: Windows 10
CPU: AMD Ryzen 5 3600X
Problem: different values for same loss and eval metric (RMSE) on same data with **categorical features** in model fitting. Python package. CatBoostRegressor model.

Code example:
```python
from catboost import CatBoostRegressor, Pool
import numpy as np
X = np.array([['en', 0, 1],
              ['es', 0, 1],
              ['en', 1, 1],
              ['en', 2, 1],
              ['ru', 4, 1],
              ['en', 1, 5],
             ])
y = [0., 0., 7., 0., 21., 0.]
model = CatBoostRegressor(loss_function="RMSE", eval_metric="RMSE", cat_features=[0])
pool = Pool(X, y, cat_features=[0])
model.fit(X, y, eval_set=pool, plot=True, verbose=False)
```
Result:
![image](https://user-images.githubusercontent.com/23147777/128379164-d97f4065-96fc-45d4-82ab-cd470bc1aa2d.png)

By removing categorical features loss and eval metric become the same
```python
from catboost import CatBoostRegressor, Pool
import numpy as np
X = np.array([[0, 0, 1],
              [1, 0, 1],
              [0, 1, 1],
              [0, 2, 1],
              [2, 4, 1],
              [0, 1, 5],
             ])
y = [0., 0., 7., 0., 21., 0.]
model = CatBoostRegressor(loss_function="RMSE", eval_metric="RMSE")
pool = Pool(X, y)
model.fit(X, y, eval_set=pool, plot=True, verbose=False)
```

Result:
![image](https://user-images.githubusercontent.com/23147777/128379623-908984ee-72d2-449c-90b9-5ac40ebea885.png)

What caused this behavior? Looking forward to your reply.

**Comment by kizill** [comment]
In your case there is relatively small amount of information in your features, and it appears, that in that particular case our inbuilt random noise added to score of binary split candidate makes CatBoost to select features with relatively low amount of information - if you set `random_strength` parameter to 0 scores would be equal.

**Comment by kizill** [comment]
Closing as solved, feel free to reopen if there are some unsolved questions.
